src/amdinfer/workers/pytorch.cpp

```diff
@@ -0,0 +1,348 @@
+#include "torch/script.h"                    // for IValue, Tensor, Device
+
+namespace fs = std::filesystem;
+
+namespace amdinfer::workers {
+
+/**
+ * @brief The PyTorchWorker worker accepts the name of an PyTorch model file as
+ * an argument and compiles and evaluates it.
+ *
+ */
+class PyTorchWorker : public SingleThreadedWorker {
+ public:
+  using SingleThreadedWorker::SingleThreadedWorker;
+  [[nodiscard]] std::vector<MemoryAllocators> getAllocators() const override;
+
+ private:
+  void doInit(ParameterMap* parameters) override;
+  void doAcquire(ParameterMap* parameters) override;
+  BatchPtr doRun(Batch* batch, const MemoryPool* pool) override;
+  void doRelease() override;
+  void doDestroy() override;
+
+  // workers define what batcher implementation should be used for them.
+  // if not explicitly defined here, a default value is used from worker.hpp.
+  // using Worker::makeBatcher;
+  // std::vector<std::unique_ptr<Batcher>> makeBatcher(
+  //   int num, ParameterMap* parameters) override {
+  //   return this->makeBatcher<HardBatcher>(num, parameters);
+  // };
+
+  // Load the model here
+  torch::jit::script::Module model_;
+
+  // Image properties
+
+  std::vector<int32_t> input_shape;
+  std::vector<int32_t> batched_input_shape;
+  uint64_t input_dim = 1;
+  std::vector<int32_t> output_shape;
+  std::vector<int32_t> batched_output_shape;
+  uint64_t output_dim = 1;
+
+  DataType input_dt_ = DataType::FP32;
+};
+
+std::vector<MemoryAllocators> PyTorchWorker::getAllocators() const {
+  return {MemoryAllocators::Cpu};
+}
+
+std::vector<int32_t> parseShape(std::string shapeStr, uint64_t dim,
+                                std::string delimiter) {
+  size_t pos = 0;
+  std::string token;
+  std::vector<int32_t> shape;
+  while ((pos = shapeStr.find(delimiter)) != std::string::npos) {
+    token = shapeStr.substr(0, pos);
+    shape.push_back(std::stoi(token));
+    shapeStr.erase(0, pos + delimiter.length());
+  }
+  shape.push_back(std::stoi(shapeStr));
+  if (shape.size() != dim) {
+    throw invalid_argument("Shapes size does not equal the dim provided.");
+  }
+  return shape;
+}
+
+void PyTorchWorker::doInit(ParameterMap* parameters) {
+  constexpr auto kBatchSize = 1;
+
+  auto batch_size = kBatchSize;
+  if (parameters->has("batch_size")) {
+    batch_size = parameters->get<int32_t>("batch_size");
+    this->batched_input_shape.push_back(batch_size);
+    this->batched_output_shape.push_back(batch_size);
+  }
+  this->batch_size_ = batch_size;
+
+  if (parameters->has("input_dim")) {
+    this->input_dim = parameters->get<int32_t>("input_dim");
+  } else {
+    throw invalid_argument(
+      "Num of dimensions of input tensors not provided in load-time "
+      "parameters");
+  }
+
+  if (parameters->has("input_shape")) {
+    std::string shape_str = parameters->get<std::string>("input_shape");
+    this->input_shape = parseShape(shape_str, this->input_dim, ",");
```

Comment:
how does this system work if there are multiple input or output tensors of different sizes? e.g. for yolo, there will be multiple output tensors

Reply:
I'm considering adding this in the next PR. 
That will also work with the added object-detection model test. These can be put in one PR.

For the implementation, I'm thinking passing the input shapes like "224,224,3;28,28,1". Tensors separated by ";" and each dim separated by ","

What do you think? Any thoughts?

Reply:
also, i'm a bit unclear about how the multiple tensors would work with the batch? 
We currently define the input PT tensor as 
```
// Initialize a PT tensor with required shape
  std::vector<int64_t> new_input_shape = {tensors};
  std::copy(this->input_shape.begin(), this->input_shape.end(),
            std::back_inserter(new_input_shape));
```
In this case, how would the PT tensor definition look like with multiple shapes?

Reply:
The PT+ZenDNN implementation you copied here did not support multiple input/output tensors so your code will need to be updated for that. This is why I wanted a test for a non object classification model to check the generality of yours.